### PR TITLE
Api branch

### PR DESCRIPTION
**Pull request by njeri-ngigi**

### CODE REVIEW

File: api/app/my-unittests.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import unittest
```
Comment (clemwek) [inline api/app/my-unittests.py]: put this file in a tests folder and rename it to `test_models.py`

File: README.md [hunk 2]
```diff
@@ -1,11 +1,13 @@
 # Hello Books API
-A RESTful Flask api for the hello-books application 
+A RESTful Flask api for the hello-books application. 
 
 <h3>Prerequisites</h3>
 
 <h3>Installation</h3>
 
 <h3>Running the tests</h3>
+Run the unittest.py tests using nosetests
```
Comment (clemwek) [inline README.md]: Explain how to and give the command you type to run the tests.

File: api/app/app.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+from models import Books, Users
+import json
+
+def create_app():   
+    app = Flask(__name__, instance_relative_config=True)
+    app.config["TESTING"] = True
+
+    book = Books()
+
+    @app.route('/books', methods=['POST', 'GET'])
```
Comment (clemwek) [inline api/app/app.py]: This is a function inside a function, I dont think that is what you want here.

File: api/tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#test_api.py
+import unittest, json
+
+from api import app 
+
+#class to respresent books testcase
+class TestCreateBooks(unittest.TestCase):
```
Comment (clemwek) [inline api/tests/test_api.py]: rename the class `TestBooks` you will have a function `test_add_books` and so on.

File: api/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
+#test_models.py
+import unittest
+from api import app
+
+#class to test models
+class TestModels(unittest.TestCase):
```
Comment (clemwek) [inline api/tests/test_models.py]: Move these functions to `test_api.py` file

File: api/app/app.py (hunk 3 above)
Comment (njeri-ngigi) [reply api/app/app.py]: This was the way I got rid of the test_client error in test_api.py

**Pull request merged**